server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -240,12 +239,15 @@ public String executor() {
         };
 
         try {
+
             // Will be replaced with ExtensionTokenProcessor and PrincipalIdentifierToken classes from feature/identity
-            final String extensionTokenProcessor = "placeholder_token_processor";
-            final String requestIssuerIdentity = "placeholder_request_issuer_identity";
 
             Map<String, List<String>> filteredHeaders = filterHeaders(headers, allowList, denyList);
 
+            TokenManager tokenManager = identityService.getTokenManager();
+            Subject subject = this.identityService.getSubject();
+            OnBehalfOfClaims claims = new OnBehalfOfClaims(discoveryExtensionNode.getId(), subject.getPrincipal().getName());
```

Comment:
Why is the issuer here the name of the current subject?

Reply:
`IdentityService.getSubject.getPrincipal.getName()` is used to grab the currently authenticated user. If we are issuing a new OBO token, I would expect it to be issued by a trigger from the active subject so this would grab that from the IdentityPlugin and then pass that  info to the OBO claims object to be made with.

Reply:
Usually the `iss` (issuer) identifies the IdP that issues the token. The `sub` is typically reserved for the principal of the subject that the token is issued for. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1

In this case, I think a good value for `iss` would be the name of the cluster from where the token was issued from.

Reply:
Oops, I must have swapped the two. I will change it.